packages/worker/src/index.ts

```diff
@@ -26,41 +27,91 @@ import {
 	MetricsService,
 	FailureMetric,
 } from '@guardian/transcription-service-backend-common/src/metrics';
+import { SQSClient } from '@aws-sdk/client-sqs';
+import { SNSClient } from '@aws-sdk/client-sns';
+
+const POLLING_INTERVAL_SECONDS = 30;
+// avoid two tasks being processed by a single worker simultaneously
+let transcriptionTaskSemaphoreTaken = false;
 
 const main = async () => {
 	const config = await getConfig();
-	const stage = config.app.stage;
-	const region = config.aws.region;
 
 	const metrics = new MetricsService(
 		config.app.stage,
 		config.aws.region,
 		'worker',
 	);
 
+	const sqsClient = getSQSClient(
+		config.aws.region,
+		config.aws.localstackEndpoint,
+	);
+	const snsClient = getSNSClient(
+		config.aws.region,
+		config.aws.localstackEndpoint,
+	);
+
+	let pollCount = 0;
+	pollTranscriptionQueue(pollCount, sqsClient, snsClient, metrics, config);
+	setInterval(() => {
+		pollCount += 1;
+		pollTranscriptionQueue(pollCount, sqsClient, snsClient, metrics, config);
+	}, POLLING_INTERVAL_SECONDS * 1000);
+};
+
+const releaseSemaphoreAndRemoveScaleInProtection = async (
+	region: string,
+	stage: string,
+) => {
+	transcriptionTaskSemaphoreTaken = false;
+	await updateScaleInProtection(region, stage, false);
+};
+
+const takeSemaphoreAndEnableScaleInProtection = async (
+	region: string,
+	stage: string,
+) => {
+	transcriptionTaskSemaphoreTaken = true;
+	await updateScaleInProtection(region, stage, true);
+};
+
+const pollTranscriptionQueue = async (
+	pollCount: number,
+	sqsClient: SQSClient,
+	snsClient: SNSClient,
+	metrics: MetricsService,
+	config: TranscriptionConfig,
+) => {
+	const stage = config.app.stage;
+	const region = config.aws.region;
 	const numberOfThreads = config.app.stage === 'PROD' ? 16 : 2;
 
-	const client = getSQSClient(region, config.aws.localstackEndpoint);
-	const message = await getNextMessage(client, config.app.taskQueueUrl);
+	if (transcriptionTaskSemaphoreTaken) {
+		console.log(`transcription task in progress. skipping`);
+		return;
+	}
+	console.log(
+		`worker polling for transcription task. Poll count = ${pollCount}`,
```

Comment:
what's a better way to say poll count? 🤔

Reply:
retry count?

Reply:
hmm I feel like retry implies a previous _failed_ attempt. Whereas polling and receiving 0 messages isn't really a failure or an error case.

Reply:
Yes that's right